services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -214,11 +220,16 @@ public void deleteNews(String newsId, org.exoplatform.services.security.Identity
     }
 
     List<String> newsTargets = newsTargetingService.getTargetsByNewsId(newsId);
-    if(newsTargets != null) {
+    if (newsTargets != null) {
       newsTargetingService.deleteNewsTargets(newsId);
     }
     newsStorage.deleteNews(newsId, isDraft);
     indexingService.unindex(NewsIndexingServiceConnector.TYPE, String.valueOf(news.getId()));
+    MetadataObject newsMetadataObject = new MetadataObject(NewsUtils.NEWS_METADATA_OBJECT_TYPE, newsId, null);
+    List<MetadataItem> metadataItems = metadataService.getMetadataItemsByObject(newsMetadataObject);
+    for (MetadataItem metadataItem : metadataItems) {
+      metadataService.deleteMetadataItem(metadataItem.getId(), true);
+    }
```

Comment:
this is not already covered by newsTargetingService.deleteNewsTargets(newsId) ?
we should keep the treatment of news targets done by newsTargetingService and not directly from newsService

Reply:
it is not for targeting news it is for the news unread activity

Reply:
ok in that case you need to delete only the metadataItems with the type unread activity ?

Reply:
It will be better to add a new method to the metadata service to delete all items by objectId, to avoid getting the list of items and delete them in a for loop.